lib/serverengine/process_manager.rb

```diff
@@ -211,7 +214,11 @@ def tick(blocking_timeout=0)
       if ready_pipes
         ready_pipes.each do |r|
           begin
-            r.read_nonblock(1024, @read_buffer)
+            if $platformwin
+              r.read(1024, @read_buffer)
```

Comment:
I suspect that child processes stall if:

* there are more than 2 child processes
* one process writes 1 byte to heatbeat pipe
* the other process writes a lot of data to heartbeat pipe (like: `loop { heatbeat_pipe.write("a"); puts "wrote" }`)